app/code/core/Mage/CatalogInventory/Model/Stock/Item.php

```diff
@@ -764,6 +764,10 @@ protected function _beforeSave()
             $this->setQty(0);
         }
 
+        if (!array_key_exists('stock_id', $this->getData())) {
```

Comment:
Think its a bit cleaner to use

    if (!$this->hasData('stock_id')) {

Reply:
Sure is, fixed.